src/graphnet/data/extractors/icecube/i3extractor.py

```diff
@@ -20,20 +25,28 @@ class I3Extractor(Extractor):
     method.
     """
 
-    def __init__(self, extractor_name: str, exclude: list = [None]):
+    def __init__(
+        self,
+        extractor_name: str,
+        exclude: list = [None],
+        is_corsika: bool = False,
+    ):
         """Construct I3Extractor.
 
         Args:
             extractor_name: Name of the `I3Extractor` instance. Used to keep
                 track of the provenance of different data, and to name tables
                 to which this data is saved.
-            exclude: List of features to exclude from the extractor.
+            exclude: List of keys to exclude from the extracted data.
+            is_corsika: Boolean indicating if the event files being processed
```

Comment:
Are there not some entry in the frame that could replace this argument?

Reply:
There are ways to check this in the frame, but then you would have to check this for every frame, and the I3Extractor itself actually does not currently do anything on the per frame level.